File: src/main/java/com/nao4j/subtrol/service/QuantityServiceImpl.java
package com.nao4j.subtrol.service;

import com.nao4j.subtrol.document.internal.Quantity.QuantityType;
import com.nao4j.subtrol.service.slice.SubscriptionSliceService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Set;

@Service
@RequiredArgsConstructor
public class QuantityServiceImpl implements QuantityService {

    private final SubscriptionSliceService sliceService;

    @Override
    public Set<QuantityType> getAll() {
        return sliceService.dimensions();
    }

}

File: src/main/kotlin/com/nao4j/subtrol/service/QuantityServiceImpl.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.model.Quantity.QuantityType
import com.nao4j.subtrol.model.Quantity.QuantityType.MONTHS
import org.springframework.stereotype.Service

@Service
class QuantityServiceImpl: QuantityService {

    override fun getAll(): Set<QuantityType> = setOf(MONTHS)

}
